src/OrbitGl/TimeGraph.h

```diff
@@ -105,10 +105,10 @@ class TimeGraph : public orbit_gl::CaptureViewElement {
                          JumpScope jump_scope);
   [[nodiscard]] const orbit_client_data::TextBox* FindPreviousFunctionCall(
       uint64_t function_address, uint64_t current_time,
-      std::optional<int32_t> thread_id = std::nullopt) const;
+      std::optional<int32_t> thread_id = std::nullopt);
```

Comment:
It's unfortunate that this is not const anymore. It seems to me like this shouldn't modify the TimeGraph in any way. What was the reason behind removing const? Same for the functions below.

Reply:
I explained it above - in TrackData::GetChains() method

Reply:
Could we add a `[[nodiscard]] *const* TimerChain* GetChain(uint64_t depth) *const*` method?

Reply:
yep - tried that, the TimerChainItartor needs to be converted to const as well otherwise we won't be able to iterate over it. And this is blocked on TextBox removal. At least it is easier to do that way. We cold introduce const and non-const iterators, but I think all iterators over chains should be const.

Reply:
As discussed offline this leads to a lot of duplication I am going to revisit it after @karupayun change.